onnxruntime/core/framework/kernel_type_str_resolver.cc

```diff
@@ -29,7 +29,9 @@ Status KernelTypeStrResolver::ResolveKernelTypeStr(const Node& node, std::string
 
   ORT_RETURN_IF(type_str_it == type_str_map.end(),
                 "Failed to find args for kernel type string '", kernel_type_str,
-                "'. If type constraint names are available, ensure that they are used in the kernel def type "
+                "' for node type '", node.OpType(), "' among ", type_str_map.size(), " registered kernels. ",
```

Comment:
`type_str_map.size()` is not the number of registered kernels. type_str_map specifies how type constraint names map to node inputs/outputs, e.g.: `{"T1": [input0, output0], "T2": [input1]}`.

I think adding the node type is useful, but I would leave out "among X registered kernels."

```suggestion
                " for '", op_id, "' node. ",
```

Reply:
done